README.md

```diff
@@ -0,0 +1,233 @@
+<!--
+---
+name: Azure Functions TypeScript Event Grid Blob Trigger using Azure Developer CLI
+description: This template repository contains an Azure Functions reference sample using the Blob trigger with Event Grid source type, written in TypeScript (v4 programming model) and deployed to Azure using the Azure Developer CLI (azd). The sample uses managed identity and a virtual network to make sure deployment is secure by default.
+page_type: sample
+products:
+- azure
+- azure-functions
+- azure-blob-storage
+- azure-virtual-networks
+- entra-id
+urlFragment: functions-quickstart-typescript-azd-eventgrid-blob
+languages:
+- typescript
+- javascript
+- nodejs
+- bicep
+- azdeveloper
+---
+-->
+
+# Azure Functions TypeScript Event Grid Blob Trigger using Azure Developer CLI
+
+This template repository contains an Azure Functions reference sample using the Blob trigger with Event Grid source type, written in TypeScript (v4 programming model) and deployed to Azure using the Azure Developer CLI (`azd`). When deployed to Azure the sample uses managed identity and a virtual network to make sure deployment is secure by default. You can control whether a VNet is used in the sample by setting `VNET_ENABLED` to true or false in the AZD parameters.
+
+This sample implements a simple function that copies PDF files from an `unprocessed-pdf` container to a `processed-pdf` container when new blobs are created. This straightforward example showcases how to use the Event Grid blob trigger to automatically respond to blob creation events in near real-time.
+
+![Architecture diagram for Azure Functions Event Grid Blob Trigger](./img/architecture.png)
+
+## Benefits of Event Grid Blob Trigger
+
+This sample uses the Event Grid source type for the Blob trigger, which provides significant advantages over the traditional scan-based approach:
+
+- **Near real-time processing**: Event Grid delivers blob events within milliseconds of creation, eliminating the delays associated with container scanning.
+- **Improved scalability**: Perfect for Flex Consumption plans where the traditional polling-based Blob trigger isn't available.
+- **Reduced costs**: Eliminates storage transaction costs from polling, which can be substantial with large numbers of blobs.
+- **Enhanced reliability**: Uses a robust pub/sub model that ensures blob events aren't missed, even during function downtime.
+- **Better performance**: No performance degradation with large numbers of blobs in a container, unlike the scan-based approach.
+
+The Event Grid approach is recommended for all new Blob trigger implementations, especially when using Flex Consumption plans where the traditional storage polling method isn't available.
+
+## Prerequisites
+
++ [Node.js 20](https://nodejs.org/)
```

Comment:
The instructions were for Node 22 not Node 20 - can you please make it Node 22? It is supported now by Flex Consumption